module1/homework/grayscaleImages/compression.hpp

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <array>
+#include <vector>
+
+constexpr size_t width = 32;
+constexpr size_t height = 32;
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(
+    const std::array<std::array<uint8_t, width>, height>&);
+
+std::array<std::array<uint8_t, width>, height> decompressGrayscale(
+    const std::vector<std::pair<uint8_t, uint8_t>>&);
+
+void printMap(const std::array<std::array<uint8_t, width>, height>&);
```

Comment:
```suggestion
using Image = std::array<std::array<uint8_t, width>, height>;

using Compressed = std::vector<std::pair<uint8_t, uint8_t>>;

Compressed compressGrayscale(const Image& compress);

Image decompressGrayscale(const Compressed& compressedMap);

void printMap(const Image& mapToPrint);
```

Reply:
Like Vic95 is suggesting you should consider aliases for longer types like std::array<std::array<uint8_t, width>, height>.